Refactor/ Restructure component layers for kebab button on containercard

ShoeheyOt opened the pull request:
<!--

Title Naming Conventions:

- Start with a capitalized prefix followed by an imperative form of a verb.
- Refer to the "Types of Branches" section for the prefix:
  https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches

Example:
- Feature: Add button to the home page
- Fix: Resolve error in data fetching
- Hotfix: Fix user authentication issue
- Refactor: Reorganize the file structure
- Test: Add unit tests for form validation
- Docs: Update installation guide

-->
## Overview
Restructuring in order to avoid nesting root in root
<!-- Provide a brief description of the changes introduced by this PR. -->

## Changes

<!-- List the changes -->
<!-- Delete this section if not needed -->
The below diagram is component (+'Root') structure,  
the left  is what it was and the right is what I changed.
As you see, I put `DialogRoot` in same level as `DropdownMenu` 

|Before|After|
|---|---|
|![containerKebabButtonbefore drawio](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/b7adbd37-7f29-4215-b275-2b8b980c9853)|![containerKebabButton drawio](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/cdb13494-e148-4d26-89ea-3aa26f96176b)|

## Screen Captures

<!-- Add screenshots or screen recordings of the changes -->
<!-- Necessary for UI changes -->
<!-- Delete this section if not needed -->
This screen capture is controled by Mouse

[KebabMenuonContainerCardwithMouse.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/06f1f804-d2b6-4adf-8c20-535ba35abbc5)

This one is by Keyboard

[KebabMenuonContainerCardwithKeyBoard.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/fe43d16b-48fa-477a-8b18-a8d9b6af137e)


## Assignee Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [x] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

nick-y-ito commented on src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/index.tsx:
```diff
@@ -1,40 +1,63 @@
    * An identifier of a container which a user is willing to delete
    */
   containerId: IContainer['id'];
+  isRenameFormOpen: boolean;
```
TSDoc for this prop. If it can be written in a single line, you can use this sysntax without a line break. Just a tip.
```ts
/** Write description here. */
isRenameFormOpen: boolean;
```

nick-y-ito commented on src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/DeleteContainerDialogContent.tsx:
This component file should not be placed here because this is not a child component that directly belongs to `ContainerCardDropdownMenuContent` but to `ContainerCardDropdownMenuTriggerButton`

nick-y-ito reviewed (changes requested):
@ShoeheyOt 
Thank you for your work! The behavior looked good! For the structure, I have left some comments on the parts that need to be revised. Kindly take a look.

ShoeheyOt commented:
@nick-y-ito  Thank you for the review! I updated what you have pointed out.
As you pointed out, I changed `ContainerCardDropdownMenuTrigger` to `ContainerCardDropdownMenu` as it is not only a trigger anymore, Please take a look below diagram how I changed the directtory structure and file name.

|before|after|
|--|--|
|![containerKebabButton-directorystructurebefore drawio](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/a2eea466-5a63-455f-b008-37b122860263)|![containerKebabButton-directorystructure drawio](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/237064e4-2f59-4972-9260-f0700588ad9d)|

nick-y-ito reviewed (approved):
@ShoeheyOt 
Thank you for the update! LTGM 😄
